gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanCompactionJobLauncher.java

```diff
@@ -42,15 +44,22 @@ public AzkabanCompactionJobLauncher(String jobId, Properties props) {
   @Override
   public void run() throws Exception {
     Class<? extends Compactor> compactorClass = getCompactorClass();
-    compactorClass.getDeclaredConstructor(Properties.class).newInstance(this.properties).compact();
+    this.compactor = compactorClass.getDeclaredConstructor(Properties.class).newInstance(this.properties);
```

Comment:
I'd move this to the constructor and make it final. Constructor should not leave the object in semi-initialized state.